File: ED-RDelAngelD-T02/Stack.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ed.rdelangeld.t02;

/**
 *
 * @author A01411440
 * @param <T>
 */
public class Stack <T>{
    private Node<T> top;
    private int size;

    public Stack() {
        this.top = null;
        this.size = 0;
    }
    public Stack(T d) {
        Node <T> node = new Node(d);
        this.top = node;
        this.size = 1;
    }

    public int getSize() {
        return size;
    }
    
    public void push(T d){
        Node <T> node = new Node(d);
        node.next = top;
        top = node;
        size++;
    }
    
    public T pop(){
        if (size>0) {
		T d = top.data;
		top = top.next;
		size--;
		return d;
	}
	else {
		return null;
	}
    }

    public T getTop() {
        if (size > 0) {
		return top.data;
	}
	else {
		return null;
	}
    }
    
    public boolean isEmpty(){
        return size==0;
    }
    
    public void showStack(){
        Node<T> t = top;
	while(t!=null) {
		t.printS();
		t = t.next;
	}
	       System.out.println("NULL");
    }
    
}

File: EstructurasLineales/C++/Stack.cpp
#include "Stack.h"

//Constructor por defecto
template <typename T>
Stack<T>::Stack() {
	top = NULL;
	size = 0;
}
//Constructor con primer elemento
template <typename T>
Stack<T>::Stack(T d) {
	Node<T> *node = new Node<T>(d);
	top = node;
	size=1;
}
//Destructor
template <typename T>
Stack<T>::~Stack() {}
//Devuelve el tamaño de la pila
template <typename T>
int Stack<T>::getSize() {
	return size;
}
//Coloca encima, recibe lo que colocara encima
template <typename T>
void Stack<T>::push(T d) {
	Node<T> *node = new Node<T>(d);
	node->next = top;
	top = node;
	size++;
}
//Elimia el valor de la cima de la pila y lo devuelve
template <typename T>
T Stack<T>::pop() {
	if (size>0) {
		T d = top->data;
		top = top->next;
		size--;
		return d;
	}
	else {
		return NULL;
	}
}

template <typename T>
T Stack<T>::getTop() {
	if (size > 0) {
		return top->data;
	}
	else {
		return NULL;
	}
}

template <typename T>
void Stack<T>::showStack() {
	Node<T> *t = top;
	while (t) {
		t->printS();
		t = t->next;
	}
	std::cout << "NULL" << endl;

}